internal/driver/glfw/device_goxjs.go

```diff
@@ -0,0 +1,37 @@
+//go:build js || wasm
+// +build js wasm
+
+package glfw
+
+import (
+	"regexp"
+	"syscall/js"
+
+	"fyne.io/fyne/v2"
+)
+
+var AndroidId = regexp.MustCompile(`Android`)
+var BlackBerryID = regexp.MustCompile(`BlackBerry`)
+var iOSID = regexp.MustCompile(`iPhone|iPad|iPod`)
+var OperaID = regexp.MustCompile(`Opera Mini`)
+var WindowsMobileID = regexp.MustCompile(`IEMobile`)
+
+var navigator = js.Global().Get("navigator")
+var userAgent = navigator.Get("userAgent").String()
+var mobileCheck = AndroidId.MatchString(userAgent) ||
+	BlackBerryID.MatchString(userAgent) ||
+	iOSID.MatchString(userAgent) ||
+	OperaID.MatchString(userAgent) ||
+	WindowsMobileID.MatchString(userAgent)
```

Comment:
Would it not be better to use `string.Contains` instead?

Reply:
I prefer MatchString here mostly for consistency and being able to group things a bit logically (I know it could be one giant match or a lot of independent Contains, kind of went in between).

Reply:
I mean, it's not just about the MatchString. I'm kind of worried about the unnecessary overhead of compiling a lot of regular expressions unnecessarily. Is it perhaps not as bad as I think?

Reply:
I don't think it could be a concern in term of speed or memory. This is called once at startup and the "compilation" of a regex is just the generation of a matrix automate. Shouldn't be much of a concern.

Reply:
Yeah that's true, but what point is there to use regex when you don't need to? Wouldn't it be just simpler and clearer to use the strings functions? It just seems to me like it is a lot of unnecessary code.

Any option? @andydotxyz

Reply:
Another thought. How about having all the device support checks in one regexp? It avoids unnecessary code clutter and would probably also avoid looping though the same content many times.

Reply:
Technically it would indeed avoid looping on the user agent string 4 times, in the entire run of the application, but I ruled that out from a maintenance and clarity point of view as this would mean mixing all the OS together with no quick glance check if a family does correctly cover a specific agent.

I do believe this piece of code is not performance critical, and that the main point of it is to be maintainable. Please review it under this angle.

Reply:
Alright then.